scipy/sparse/_construct.py

```diff
@@ -690,29 +769,29 @@ def bmat(blocks, format=None, dtype=None):
     M,N = blocks.shape
 
     # check for fast path cases
-    if (format in (None, 'csr') and all(isinstance(b, csr_matrix)
-                                        for b in blocks.flat)):
+    if (format in (None, 'csr') and
+        all(issparse(b) and b.format == "csr" for b in blocks.flat)
```

Comment:
This question is also bigger than this issue.  I know that you prefer type checking rather than attribute checking. But that is not universal and I'm just going with what I understand is the standard way to check.  The current form will not require any change after `csr_matrix` is removed. (I don't know the other parts of that discussion.)